File: web/src/api/resolve.ts
import client from './client'

export interface ResolvedContext {
  owner_id: string
  companion_id: string
  device_id: string
  memory_realm_id: string
  genome_id: string
  schema_version: string
  genome_hash: string
  compiler_version: string
  interaction_mode: string | null
}

export interface ResolveDeviceResponse {
  context: ResolvedContext
}

export async function resolveDevice(deviceId: string): Promise<ResolvedContext> {
  const { data } = await client.get<ResolveDeviceResponse>(
    `/resolve/device/${encodeURIComponent(deviceId)}`,
  )
  return data.context
}

File: server/eidolon_admin_server/app/registry/schemas/resolve.py
"""Resolve device-bound runtime identity for callers."""

from __future__ import annotations

from pydantic import BaseModel


class ResolvedContext(BaseModel):
    """Everything a runtime session needs in one envelope."""

    owner_id: str
    companion_id: str
    device_id: str
    memory_realm_id: str
    genome_id: str
    interaction_mode: str | None = None


class ResolveDeviceResponse(BaseModel):
    """GET /api/resolve/device/{device_id} response."""

    context: ResolvedContext
